src/promptflow-tools/promptflow/tools/azure_content_safety.py

```diff
@@ -0,0 +1,319 @@
+        TextCategorySensitivity.HIGH_SENSITIVITY: 2,
+    }
+    return switcher.get(sensitivity, f"Non-supported sensitivity: {sensitivity}")
+
+
+class MediaType(enum.Enum):
+    Text = 1
+    Image = 2
+
+
+class Category(enum.Enum):
+    Hate = 1
+    SelfHarm = 2
+    Sexual = 3
+    Violence = 4
+
+
+class Action(enum.Enum):
+    Accept = "Accept"
+    Reject = "Reject"
+
+
+class DetectionError(Exception):
+    def __init__(self, code: str, message: str) -> None:
+        """
+        Exception raised when there is an error in detecting the content.
+
+        Args:
+        - code (str): The error code.
+        - message (str): The error message.
+        """
+        self.code = code
+        self.message = message
+
+    def __repr__(self) -> str:
+        return f"DetectionError(code={self.code}, message={self.message})"
+
+
+class Decision(object):
+    def __init__(self, suggested_action: Action, action_by_category: Dict[Category, Action]) -> None:
+        """
+        Represents the decision made by the content moderation system.
+
+        Args:
+        - suggested_action (Action): The suggested action to take.
+        - action_by_category (dict[Category, Action]): The action to take for each category.
+        """
+        self.suggested_action = suggested_action
+        self.action_by_category = action_by_category
+
+
+def convert_decision_to_json(decision: Decision):
+    result_json = {}
+    result_json["suggested_action"] = decision.suggested_action.value
+    category_json = {}
+    for key, value in decision.action_by_category.items():
+        category_json[key.name] = value.value
+    result_json["action_by_category"] = category_json
+    return result_json
+
+
+class ContentSafety(object):
+    def __init__(self, endpoint: str, subscription_key: str, api_version: str) -> None:
+        """
+        Creates a new ContentSafety instance.
+
+        Args:
+        - endpoint (str): The endpoint URL for the Content Safety API.
+        - subscription_key (str): The subscription key for the Content Safety API.
+        - api_version (str): The version of the Content Safety API to use.
+        """
+        self.endpoint = endpoint
+        self.subscription_key = subscription_key
+        self.api_version = api_version
+
+    def build_url(self, media_type: MediaType) -> str:
+        """
+        Builds the URL for the Content Safety API based on the media type.
+
+        Args:
+        - media_type (MediaType): The type of media to analyze.
+
+        Returns:
+        - str: The URL for the Content Safety API.
+        """
+        if media_type == MediaType.Text:
+            return f"{self.endpoint}/contentsafety/text:analyze?api-version={self.api_version}"
+        elif media_type == MediaType.Image:
+            return f"{self.endpoint}/contentsafety/image:analyze?api-version={self.api_version}"
+        else:
+            raise ValueError(f"Invalid Media Type {media_type}")
```

Comment:
Create your own specific exception for your tool like errors in https://github.com/microsoft/promptflow/blob/main/src/promptflow-tools/promptflow/tools/exception.py

Reply:
And please test if you have an invalid type, if flow throws the correct exception in vscode.